src/main/java/org/folio/circulation/resources/ItemsLostRequiringActualCostResource.java

```diff
@@ -0,0 +1,174 @@
+      .thenComposeAsync(repositories.getOverdueFinePolicy()::findOverdueFinePolicyForLoan)
+      .thenComposeAsync(repositories.getLostItemPolicy()::findLostItemPolicyForLoan)
+      .thenComposeAsync(repositories.getPatronGroup()::findGroupForLoan)
+      .thenApply(r -> r.map(loan -> entry.withLoan(loan)));
+  }
+
+  private CompletableFuture<Result<JsonObject>> mapEntryToJson(ItemLostRequiringCostsEntry entry,
+    ItemLostRequiringActualCostsRepresentation itemRepresentation) {
+
+    if (entry.getLoan() != null && entry.getLoan().isOpen()) {
+      if (!entry.getItem().getStatusName().equals(AGED_TO_LOST.getValue()) ||
+        entry.getLoan().getLostItemHasBeenBilled()) {
+
+        if (hasNoLostFeeFineActualCosts(entry.getLoan())) {
+          return Result.ofAsync(() -> itemRepresentation.mapToResult(entry));
+        }
+      }
+    }
+
+    return Result.ofAsync(() -> null);
+  }
+
+  private boolean hasNoLostFeeFineActualCosts(Loan loan) {
+    if (loan.getAccounts() != null && loan.getAccounts().size() > 0) {
+      for (Account account : loan.getAccounts()) {
+        if (account.getFeeFineType().equals(FeeFine.LOST_ITEM_FEE_ACTUAL_COSTS_TYPE)) {
+          return false;
+        }
+      }
+    }
+
+    return true;
+  }
+
+  private Result<JsonObject> mapResultToJson(Result<List<JsonObject>> items) {
+    Result<JsonArray> result = items.map(r -> r.stream()
+      .filter(Objects::nonNull)
+      .collect(Collector.of(JsonArray::new, JsonArray::add, JsonArray::add)));
+
+    return result.next(jsonArray -> Result.succeeded(new JsonObject()
+      .put("items", jsonArray)
+      .put("totalRecords", jsonArray.size())));
+  }
+
+  private CompletableFuture<Result<Loan>> fetchLoanUser(Result<Loan> result, UserRepository userRepository) {
+    return result.combineAfter(userRepository::getUser, Loan::withUser);
+  }
+
+  @AllArgsConstructor
+  @Getter
+  private class Repositories {
+    private final ItemRepository item;
+    private final LoanRepository loan;
+    private final ServicePointRepository servicePoint;
+    private final UserRepository user;
+    private final LoanPolicyRepository loanPolicy;
+    private final OverdueFinePolicyRepository overdueFinePolicy;
+    private final LostItemPolicyRepository lostItemPolicy;
+    private final AccountRepository account;
+    private final PatronGroupRepository patronGroup;
+    private final FeeFineActionRepository feeFineAction;
+
+    public Repositories(Clients clients) {
+      this(
+        new ItemRepository(clients, true, true, true),
+        new LoanRepository(clients),
+        new ServicePointRepository(clients),
+        new UserRepository(clients),
+        new LoanPolicyRepository(clients),
+        new OverdueFinePolicyRepository(clients),
+        new LostItemPolicyRepository(clients),
+        new AccountRepository(clients),
+        new PatronGroupRepository(clients),
+        new FeeFineActionRepository(clients)
+      );
```

Comment:
Following feedback from others, I stopped putting each parameter on each line. Do you think this reads better / only one fits on a line anyway?